google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/migration/StagedArtifactNameConverterProvider.java

```diff
@@ -0,0 +1,125 @@
+import com.intellij.conversion.ConversionProcessor;
+import com.intellij.conversion.ConverterProvider;
+import com.intellij.conversion.ProjectConverter;
+import com.intellij.conversion.RunManagerSettings;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+import org.jdom.Attribute;
+import org.jdom.Element;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * A {@link ConverterProvider} for converting the staged artifact name in App Engine run
+ * configurations.
+ *
+ * <p>With the introduction of this field, existing run configurations may rely on the previous
+ * behavior for staging artifacts for custom runtime deployments. This behavior was to always rename
+ * the built artifact to "target.jar" or "target.war", depending on the type of artifact that was
+ * built. This conversion explicitly sets this new field to the expected default value so these
+ * existing run configurations continue to operate correctly.
+ */
+public final class StagedArtifactNameConverterProvider extends ConverterProvider {
+
+  public StagedArtifactNameConverterProvider() {
+    super("gcp-staged-artifact-name-converter");
+  }
+
+  @NotNull
+  @Override
+  public String getConversionDescription() {
+    return GctBundle.message("appengine.staged.artifact.name.converter.description");
+  }
+
+  @NotNull
+  @Override
+  public ProjectConverter createConverter(@NotNull ConversionContext context) {
+    return new ProjectConverter() {
+      @Override
+      public ConversionProcessor<RunManagerSettings> createRunConfigurationsConverter() {
+        return new StagedArtifactNameConversionProcessor();
+      }
+    };
+  }
+
+  /** A {@link ConversionProcessor} for converting the staged artifact name. */
+  private static final class StagedArtifactNameConversionProcessor
+      extends ConversionProcessor<RunManagerSettings> {
+
+    private static final Function<Element, Stream<Element>> TO_DEPLOYMENT_SETTINGS =
+        element -> {
+          Element deployment = element.getChild("deployment");
+          if (deployment != null) {
+            Element settings = deployment.getChild("settings");
+            if (settings != null) {
+              return Stream.of(settings);
+            }
+          }
+          return Stream.empty();
+        };
+
+    private static final Predicate<Element> IS_STAGED_ARTIFACT_NAME_NULL =
+        element -> element.getAttribute(STAGED_ARTIFACT_NAME) == null;
+
+    private static final Predicate<Element> HAS_ARTIFACT_PATH =
+        element -> element.getAttribute(USER_SPECIFIED_ARTIFACT_PATH_ATTRIBUTE) != null;
+
+    @Override
+    public boolean isConversionNeeded(RunManagerSettings runManagerSettings) {
+      return runManagerSettings
+          .getRunConfigurations()
+          .stream()
+          .flatMap(TO_DEPLOYMENT_SETTINGS)
+          .anyMatch(IS_STAGED_ARTIFACT_NAME_NULL.and(HAS_ARTIFACT_PATH));
```

Comment:
isn't the goal of the migrator to determine if its an "old" run config (e.g. one that was created prior to these change) and then, if so, default the artifact field to "target.jar/war"?

If its determined _not_ to be an old configuration then the field can be null/empty which indicates that we should use the real artifact in the deployment (which can be populated later, not here in the migration).

Am I missing something?